fixing error when istgt is started with no volume

mynktl opened the pull request:
Changes : 
- If `ISTGT` is started without any volume configuration then replication module will not accept connection from a replica.
- In `replication_test`, if it fails to make a connection with the target then it will exit with return value `1`.

Commit:
initial commit

Signed-off-by: mayank <mayank.patel@cloudbyte.com>

Commit:
Resolve problems when istgt is started with no volume

Signed-off-by: mayank <mayank.patel@cloudbyte.com>

vishnuitta commented on src/istgt_lu.h:
```diff
@@ -326,6 +328,7 @@ typedef struct istgt_lu_t {
 	ISTGT_LU_MAP map[MAX_LU_MAP];
 	int conns;
 #ifdef REPLICATION
+	uint64_t replica_id[MAXREPLICA];
```
why is this required?

mynktl replied:
ok. My base branch was different. That's why this change is here. I will remove this change.

vishnuitta commented on src/replication.c:
```diff
@@ -1453,6 +1453,12 @@ accept_mgmt_conns(int epfd, int sfd)
 
+		if (!spec) {
+			REPLICA_ERRLOG("Spec is not configured\n");
+			close(mgmt_fd);
```
you need to do 'shutdown' on fd also to inform other side of fd

mynktl replied:
ok

vishnuitta commented on src/replication_test.c:
```diff
@@ -550,6 +550,11 @@ main(int argc, char **argv)
 					}
 					break;
 				}
+				if (count == 0) {
```
are we hitting this when just 'close' is done on other side?

mynktl replied:
No.  Actually, in `test_read_data`, we are not handling return value `0` from `read`. Due to this, have added this check in the caller of `test_read_data`.

vishnuitta commented on src/replication_test.c:
```diff
@@ -614,6 +622,9 @@ main(int argc, char **argv)
 						} else if(((uint64_t)count < total_len - recv_len) && errno == EAGAIN) {
 							recv_len += count;
 							break;
+						} else if (count == 0) {
```
this should be after initial 'if' check for count < 0..
same thing with next chunk of changes also

mynktl replied:
ok

vishnuitta reviewed (commented):
testcase need to be added, like,
- bring up with no LU
- connect with replicas
- add LU section and do refresh
- connect should be done automatically

Commit:
Adding test case for lun refresh
In test(run_lu_add_test) :
 - ISTGT will start without any LUN configuration.
 - Starts replica with '-r' argument so that replica will try
   to establish the connection with target again and again until it gets successfully connected
 - Update istgt.conf with LUN configuration and execute 'istgtcontrol refresh'
 - now, the replica will be able to connect target successfully

Signed-off-by: mayank <mayank.patel@cloudbyte.com>

Commit:
Merge remote-tracking branch 'upstream/replication' into TA823

Signed-off-by: mayank <mayank.patel@cloudbyte.com>

Conflicts:
	src/replication_test.c
	test_istgt.sh

Commit:
fixing travis build error

Signed-off-by: mayank <mayank.patel@cloudbyte.com>

vishnuitta reviewed (approved):
changes are good
